src/util/parse_options.cpp

```diff
@@ -65,3 +66,31 @@ int parse_options_baset::main()
 
   return doit();
 }
+
+int parse_optionst::main()
+{
+  // catch all exceptions here so that this code is not duplicated
+  // for each tool
+  try
+  {
+    return parse_options_baset::main();
+  }
+  catch(const system_exceptiont &e)
+  {
+    message.error() << e.what() << messaget::eom;
+  }
+  catch(const input_src_exceptiont &e)
+  {
+    message.error().source_location=e.source_location;
+    message.error() << e.what() << messaget::eom;
+  }
+  catch(const ui_exceptiont &e)
+  {
+    message.error() << e.what() << messaget::eom;
+  }
+  catch(const invariant_failedt &e)
+  {
+    message.error() << e.what() << messaget::eom;
+  }
+  return 6;
```

Comment:
Since all these exceptions are fatal and shouldn't be caught anywhere else, I wonder whether we should include the exit code in the exceptions. Then the different catch clauses would return via `return e.exit_code()`.

Reply:
Yes, that could make some sense.